php-checks/src/main/java/org/sonar/php/checks/UnusedPrivateMethodCheck.java

```diff
@@ -80,12 +87,38 @@ private void checkClass(ClassTree tree) {
         && !dynamicUsedMethods.contains(methodSymbol.name())
         && !isConstructor(methodSymbol.declaration(), tree)
         && !isMagicMethod(methodSymbol.name())
-        && !isUsedInStringLiteral(methodSymbol)) {
+        && !isUsedInStringLiteral(methodSymbol)
+        && !isMagicMethodCallDefined(tree)) {
         context().newIssue(this, methodSymbol.declaration(), String.format(MESSAGE, methodSymbol.name()));
       }
     }
   }
 
+  private static boolean isMagicMethodCallDefined(ClassTree tree) {
+    if (tree instanceof ClassDeclarationTree) {
+      Optional<ClassMemberTree> magicMethodCall = findMagicMethodCall((ClassDeclarationTree) tree);
```

Comment:
How should we proceed with `__call` defined in a superclass? I believe this will also have the same behavior. Are we able to check it or should it be considered too rare to bother?

Reply:
Good point. 
Now it checks if a `__call` method is defined in one of the superclasses. A more detailed check (if there `call_user_func*` is called) is not possible with the current symbol table implementation.